javascript/javascript1/week4/appUsage.js

```diff
@@ -0,0 +1,71 @@
+
+//max usage of activity in day
+const usage = [];
+const activityUsage = [];
+const usageNew = [];
+const durationArr = [];
+
+function addActivity(activity, duration) 
+{
+    const today = new Date();
+    usage.push({
+        date: today.toLocaleDateString("en-US"),
+        activity: activity,
+        duration: duration
+    })
+}
+addActivity("facebook", 10);
+addActivity("twitter", 15);
+addActivity("linked-in", 20);
+addActivity("youtube", 30);
+addActivity("twitter", 25);
+addActivity("whatsapp", 16);
+addActivity("youtube", 20);
+addActivity("twitter", 5);
+console.log(usage);
+
+function aggrigateActivities() 
+{
+    const uniqueActivity = [];
+    for (let i = 0; i < usage.length; i++) 
+    {
+        let activity = usage[i].activity;
+        let duration = usage[i].duration;
+        if (! uniqueActivity.includes(activity))
+        {
+            uniqueActivity.push(activity);
+            for (let j = i + 1; j < usage.length; j++) 
+            {
+                if (activity === usage[j].activity)
+                {
+                    duration = duration + usage[j].duration;
+                }
+            }
+            activityUsage.push({activity, duration});
+        }
+    }
+    console.log(uniqueActivity);
+    console.log(activityUsage);
+}
+
+function maxUsage()
+{
+    let maxkey = "";
+    let maxValue = 0
+    for (let i = 0; i < activityUsage.length; i++) 
+    {
+        if (activityUsage[i].duration > maxValue)
+        {
+            maxkey = activityUsage[i].activity;
+            maxValue = activityUsage[i].duration;
+        }
+    }
+    return `The maximum used of ${maxkey} and you used ${maxValue} min`;
+}
+
+aggrigateActivities();
+const arr = maxUsage();
+console.log(arr);
+
```

Comment:
Instead of mutating global state consider returning the result from the function `aggregateActivities(activities)` and storing it into a variable.

then using the result to attain the activity with the most time spent.

Something like this:

```js
function getMostFrequentActivity(activities){
      var aggregatedActivities = aggregateActivitiesDuration(activities);
      var activityWithHighestDuration = getActivityWithHighestDuration(aggregatedActivities);
      return activityWithHighestDuration;
}

function getActivityWithHighestDuration(activities) { //return activity with highest duration}

function aggregateActivitiesDuration(activities) {//return activities with aggregated duration by activity name}

var mostFrequentActivity = getMostFrequentActivity(aggregatedActivities);
console.log(`The maximum used of ${mostFrequentActivity.activity} and you used ${mostFrequentActivity.duration} min`)
```

Reply:
Thank you for giving me feedback. it was very helpful and liked it. I will use in further coding assignments